File: nodejs/checkio/src/initiation/remove_all_before.ts
import assert from "assert";

function removeAllBefore(values: number[], b: number): number[] {
	return values.slice(values.indexOf(b) === -1 ? 0 : values.indexOf(b));
}

console.log("Example:");
console.log(JSON.stringify(removeAllBefore([1, 2, 3, 4, 5], 3)));

// These "asserts" are used for self-checking
assert.deepStrictEqual(removeAllBefore([1, 2, 3, 4, 5], 3), [3, 4, 5]);
assert.deepStrictEqual(removeAllBefore([1, 1, 2, 2, 3, 3], 2), [2, 2, 3, 3]);
assert.deepStrictEqual(
	removeAllBefore([1, 1, 2, 4, 2, 3, 4], 2),
	[2, 4, 2, 3, 4],
);
assert.deepStrictEqual(removeAllBefore([1, 1, 5, 6, 7], 2), [1, 1, 5, 6, 7]);
assert.deepStrictEqual(removeAllBefore([], 0), []);
assert.deepStrictEqual(
	removeAllBefore([7, 7, 7, 7, 7, 7, 7, 7, 7], 7),
	[7, 7, 7, 7, 7, 7, 7, 7, 7],
);

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/initiation/remove-all-before.py
from collections.abc import Iterable


def remove_all_before(items: list, border: int) -> Iterable:
    return items[items.index(border) :] if border in items else items


print("Example:")
print(list(remove_all_before([1, 2, 3, 4, 5], 3)))

# These "asserts" are used for self-checking
assert list(remove_all_before([1, 2, 3, 4, 5], 3)) == [3, 4, 5]
assert list(remove_all_before([1, 1, 2, 2, 3, 3], 2)) == [2, 2, 3, 3]
assert list(remove_all_before([1, 1, 2, 4, 2, 3, 4], 2)) == [2, 4, 2, 3, 4]
assert list(remove_all_before([1, 1, 5, 6, 7], 2)) == [1, 1, 5, 6, 7]
assert list(remove_all_before([], 0)) == []
assert list(remove_all_before([7, 7, 7, 7, 7, 7, 7, 7, 7], 7)) == [
    7,
    7,
    7,
    7,
    7,
    7,
    7,
    7,
    7,
]

print("The mission is done! Click 'Check Solution' to earn rewards!")
